src/Raven.Server/Documents/Subscriptions/Sharding/ShardDocumentSubscriptionFetcher.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections.Generic;
+
+namespace Raven.Server.Documents.Subscriptions.Sharding;
+
+public class ShardDocumentSubscriptionFetcher : DocumentSubscriptionFetcher
+{
+    public ShardDocumentSubscriptionFetcher(DocumentDatabase database, SubscriptionConnectionsState subscriptionConnectionsState, string collection) :
+        base(database, subscriptionConnectionsState, collection)
+    {
+    }
+
+    protected override IEnumerable<Document> FetchFromResend()
+    {
+        var records = new SortedDictionary<long, Document>();
+        foreach (var document in base.FetchFromResend())
+        {
+            var current = Database.DocumentsStorage.GetDocumentOrTombstone(DocsContext, document.Id, throwOnConflict: false);
+            using (current.Document)
+            using (current.Tombstone)
+            {
+                if (current.Missing || current.Document == null)
+                {
+                    document.Dispose();
+                    // items from another shard or might be already deleted
+                    continue;
+                }
+
+                records.Add(current.Document.Etag, document);
```

Comment:
you read the entire document only to get the etag?

Reply:
I've pushed some adjustments, @karmeli87 @garayx please review

Reply:
Not sure why `GetDocumentOrTombstone` method was used. Nothing was done with the tombstone

Reply:
looks fine, I used `GetDocumentOrTombstone` method since it was used in `ShouldSend()` in any case I dropped its usage from there as well.